File: RoomDatabaseTestProject/app/src/main/java/com/example/roomdatabasetestproject/database/dao/StudentDao.java
package com.example.roomdatabasetestproject.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.roomdatabasetestproject.database.entitiy.Student;

import java.util.List;

@Dao
public interface StudentDao {

    @Query("SELECT * FROM Student")
    List<Student> getAllStudents();

    @Query("SELECT * FROM Student s WHERE s.studentId = :id")
    Student getStudentById(Integer id);

    @Insert
    void insertStudent(Student student);

    @Delete
    void deleteStudent(Student student);

    @Update
    void updateStudent(Student student);
}

File: app/src/main/java/com/example/teachtech/database/dao/StudentDao.kt
package com.example.teachtech.database.dao;
import androidx.room.*
import com.example.teachtech.database.entity.Student


@Dao
interface StudentDao {

    @get:Query("SELECT * FROM Student")
    val allStudents: List<Student?>?

    @Query("SELECT * FROM Student s WHERE s.studentId = :id")
    fun getStudentById(id: Int?): Student?

    @Insert
    fun insertStudent(student: Student?)

    @Delete
    fun deleteStudent(student: Student?)

    @Update
    fun updateStudent(student: Student?)
}